File: code/Python/Chapter12/12-3/path.py
def path(row, col):
    path = [[1] * col] + [[1] + [0] * (col - 1) for _ in range(row - 1)]
    for i in range(1, row):
        for j in range(1, col):
            path[i][j] = path[i - 1][j] + path[i][j - 1]
    return path[row - 1][col - 1]

print(path(4, 8))

File: code/Java/Chapter12/12-3/Path.java
public class Path {
    public static void main(String[] args) {
        System.out.println(path(4, 8));
    }

    public static int path(int row, int col) {
        int[][] path = new int[row][n];
        for (int i = 0; i < row; i++) {
            for (int j = 0; j < col; j++) {
                if (i == 0 || j == 0) {
                    path[i][j] = 1;
                } else {
                    path[i][j] = path[i - 1][j] + path[i][j - 1];
                }
            }
        }
        return path[row - 1][col - 1];
    }
}
